integration-tests/suites/k8s_namespace.go

```diff
@@ -0,0 +1,270 @@
+package suites
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/stackrox/collector/integration-tests/pkg/collector_manager"
+	"github.com/stackrox/collector/integration-tests/pkg/executor"
+
+	coreV1 "k8s.io/api/core/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+const (
+	NAMESPACE = "target-namespace"
+)
+
+type NamespaceTest struct {
+	containerID       string
+	expectecNamespace string
+}
+
+type K8sNamespaceTestSuite struct {
+	IntegrationTestSuiteBase
+	tests       []NamespaceTest
+	collectorIP string
+}
+
+func (k *K8sNamespaceTestSuite) SetupSuite() {
+	// Ensure the collector pod gets deleted
+	k.T().Cleanup(func() {
+		exec, err := k.Executor()
+		k.Require().NoError(err)
+
+		exists, err := exec.ContainerExists(executor.PodFilter{
+			Name:      "collector",
+			Namespace: collector_manager.TEST_NAMESPACE,
+		})
+		k.Require().NoError(err)
+		if exists {
+			collector, err := k.Collector()
+			k.Require().NoError(err)
+			collector.TearDown()
+		}
+
+		if k.sensor != nil {
+			k.sensor.Stop()
+		}
+
+		nginxPodFilter := executor.PodFilter{
+			Name:      "nginx",
+			Namespace: NAMESPACE,
+		}
+		exists, err = exec.ContainerExists(nginxPodFilter)
+		k.Require().NoError(err)
+
+		if exists {
+			exec.RemoveContainer(nginxPodFilter)
+		}
+	})
+
+	// Start Sensor
+	k.Sensor().Start()
+
+	collector, err := k.Collector()
```

Comment:
The error here is ignored

Reply:
This one is gone after removing the need for `k.Collector()` to return an error.